src/Ui/RestGetTicket.php

```diff
@@ -25,32 +31,76 @@
 use Horde_Session;
 use Horde\Log\Logger;
 
-
-
-
 /**
  * Controller class for Public UI
  */
 class RestGetTicket extends RestBase
 {
-    
-    /**
-     * Overload this method for actually implementing stuff
-     * 
-     * Quick & Dirty: Use $this->injector to get application services
-     * Proper: Overload and amend constructor
-     */
+    protected function getResponseStream(array $data)
+    {
+        return $this->streamFactory->createStream(json_encode($data));
+    }
+
     protected function buildResponseStream(ServerRequestInterface $request): ?StreamInterface
```

Comment:
Same here, if you provide no default implementation that is either useful or illustrational, declare the base as abstract.